src/frontend/Semantic_error.ml

```diff
@@ -31,6 +31,11 @@ module TypeError = struct
         * UnsizedType.t list
         * (UnsizedType.autodifftype * UnsizedType.t) list
         * SignatureMismatch.function_mismatch
+    | IllTypedVariadicDAE of
+        string
+        * UnsizedType.t list
+        * (UnsizedType.autodifftype * UnsizedType.t) list
+        * SignatureMismatch.function_mismatch
```

Comment:
I would personally rather not create a new error type here and instead just make the current `IllTypedVariadicODE` into `IllTypedVariadicFn`. The type will need to include the return type (which will be either                       Stan_math_signatures.variadic_dae_fun_return_type or                       Stan_math_signatures.variadic_ode_fun_return_type) but it saves the duplication here

Reply:
Thanks. I just changed replace both with `IllTypedVariadicDE` with an additional arg for return type.